plugins/catalog/src/utils/getEntitySourceLocation.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Entity,
+  parseLocationReference,
+  SOURCE_LOCATION_ANNOTATION,
+} from '@backstage/catalog-model';
+import { ConfigApi } from '@backstage/core';
+import { ScmIntegrations } from '@backstage/integration';
+
+export type EntitySourceLocation = {
+  url: string;
+  type?: string;
+};
+
+export function getEntitySourceLocation(
+  entity: Entity,
+  config: ConfigApi,
+): EntitySourceLocation | undefined {
+  const sourceLocation =
+    entity.metadata.annotations?.[SOURCE_LOCATION_ANNOTATION];
+
+  if (!sourceLocation) {
+    return undefined;
+  }
+
+  try {
+    const sourceLocationRef = parseLocationReference(sourceLocation);
+    const scmIntegrations = ScmIntegrations.fromConfig(config);
```

Comment:
This makes me realise - i wonder if this constructor actually is robust for all providers _on the frontend_, or if it will reject things (perhaps even throw) as it does not find any tokens / passwords etc that are only available in the backend because of the config system stripping them out on the frontend...

Also nothing that this is pretty unfortunate to have to construct over and over especially in a large table or so. It should be available as `useIntegrations` basically and passed in by the component. But OK, let's do that separately.

Reply:
Hmm, yeah. Having that cached would be neat.